src/allmydata/test/common_system.py

```diff
@@ -700,6 +700,8 @@ def setUp(self):
         self.sparent.startService()
 
     def _got_new_http_connection_pool(self, pool):
+        # Make sure the pool closes cached connections quickly:
+        pool.cachedConnectionTimeout = 0.1
```

Comment:
My hypothesis is that a connection was getting added back to the pool _after_ `closeCachedConnections()` was called, and then never getting cleaned up because the timeout was 240 seconds.